server.js

```diff
@@ -1,21 +1,81 @@
+// with a user object, which will be set at `req.user` in route handlers after
+// authentication.
+passport.use(new Strategy({
+    clientID: '205751546505611',
+    clientSecret: 'e78708edd0c07fc11dcca84010cfc433',
+    callbackURL: 'http://localhost:3000/login/facebook/return',
+    profileFields: ['id', 'displayName', 'email', 'picture', 'friends']
+  },
+  function(accessToken, refreshToken, profile, cb) {
+    // In this example, the user's Facebook profile is supplied as the user
+    // record.  In a production-quality application, the Facebook profile should
+    // be associated with a user record in the application's database, which
+    // allows for account linking and authentication with other identity
+    // providers.
+    return cb(null, profile);
+  }));
+
+
+// Configure Passport authenticated session persistence.
+//
+// In order to restore authentication state across HTTP requests, Passport needs
+// to serialize users into and deserialize users out of the session.  In a
+// production-quality application, this would typically be as simple as
+// supplying the user ID when serializing, and querying the user record by ID
+// from the database when deserializing.  However, due to the fact that this
+// example does not have a database, the complete Twitter profile is serialized
+// and deserialized.
+passport.serializeUser(function(user, cb) {
+  cb(null, user);
+});
+
+passport.deserializeUser(function(obj, cb) {
+  cb(null, obj);
+});
+
 app.set('view engine', 'html');
 app.engine('html', require('ejs').__express);
-
 app.use(express.static(__dirname + '/public'));
 app.use('/node_modules', express.static(__dirname + '/node_modules'));
 app.use(bodyParser.urlencoded({ extended: true }));
+app.use(morgan('combined'));
+app.use(cookieParser());
+app.use(expressSession({ secret: 'keyboard cat', resave: true, saveUninitialized: true }));
+
+// Initialize Passport and restore authentication state, if any, from the
+// session.
+app.use(passport.initialize());
+app.use(passport.session());
 
 const events = {};
 
-app.get('/', (req, res) => res.render('home'));
-app.get('/events', (req, res) => res.render('events', { events }));
-app.get('/events/:id', (req, res) => res.render('edit', { event: events[req.params.id] }));
-app.get('/events/:id/share', (req, res) => res.render('share', { event: events[req.params.id] }));
+app.get('/', (req, res) => res.render('home', { user: req.user }));
+
+app.get('/login', (req, res) => res.render('login', { user: req.user }));
+
+app.get('/login/facebook', passport.authenticate('facebook', { scope: ['user_friends', 'email'] }));
+
+app.get('/login/facebook/return', passport.authenticate('facebook', { failureRedirect: '/login/facebook' }), (req, res) => res.redirect('/events'));
+
+//app.get('/events', (req, res) => res.render('events', { events }));
+app.get('/events', require('connect-ensure-login').ensureLoggedIn(), (req, res) => res.render('events', { user: req.user, events }));
```

Comment:
I cannot protect app.post('/events') because otherwise we would need to change the way how we create an event. 

i.e. right now it does: home page -> POST events  -> redirect To events/id -> not authenticated -> redirect to login -> login -> redirect to events

it should be something like:  home page -> POST events -> not authenticated -> redirect to login -> login -> redirect to events -> do the real POST -> redirect to events/id

do you see what i mean?